File: app/src/main/java/com/commit451/gitlab/adapter/ProjectsPagerAdapter.java
package com.commit451.gitlab.adapter;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.commit451.gitlab.R;
import com.commit451.gitlab.fragment.ProjectsFragment;

/**
 * Projects Pager Adapter
 * Created by Jawn on 9/21/2015.
 */
public class ProjectsPagerAdapter extends FragmentPagerAdapter {
    private static final int SECTION_COUNT = 3;

    private String[] mTitles;

    public ProjectsPagerAdapter(Context context, FragmentManager fm) {
        super(fm);
        mTitles = context.getResources().getStringArray(R.array.projects_tabs);
    }

    @Override
    public Fragment getItem(int position) {

        switch(position) {
            case 0:
                return ProjectsFragment.newInstance(ProjectsFragment.MODE_ALL);
            case 1:
                return ProjectsFragment.newInstance(ProjectsFragment.MODE_MINE);
            case 2:
                return ProjectsFragment.newInstance(ProjectsFragment.MODE_STARRED);
        }

        throw new IllegalStateException("Position exceeded on view pager");
    }

    @Override
    public int getCount() {
        return SECTION_COUNT;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mTitles[position];
    }
}

File: app/src/main/java/com/commit451/gitlab/adapter/ProjectsPagerAdapter.kt
package com.commit451.gitlab.adapter

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.commit451.gitlab.R
import com.commit451.gitlab.fragment.ProjectsFragment

/**
 * Projects Pager Adapter
 */
class ProjectsPagerAdapter(context: Context, fm: FragmentManager) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    private val titles: Array<String> = context.resources.getStringArray(R.array.projects_tabs)

    override fun getItem(position: Int): Fragment {

        when (position) {
            0 -> return ProjectsFragment.newInstance(ProjectsFragment.MODE_ALL)
            1 -> return ProjectsFragment.newInstance(ProjectsFragment.MODE_MINE)
            2 -> return ProjectsFragment.newInstance(ProjectsFragment.MODE_STARRED)
        }

        throw IllegalStateException("Position exceeded on view pager")
    }

    override fun getCount(): Int {
        return titles.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return titles[position]
    }
}
